File: Jaewon/17) 소수 찾기/17.js
function solution(n) {
    var answer = 0;
    for(let i = n; i > 1; i--){
        if(i%2 === 0) continue;
        if(isPrime(i)) answer++;
    }
    return answer+1;
}

function isPrime(n){
    for(let i = 2; i*i <= n; i++){
        if(n%i == 0) return false;
    }
    return true;
}

File: Jaewon/17) 소수 찾기/17.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 1;
    vector<bool> arr(n+1, true);

    for (int i=2; i<=n; i++) {
        if (i%2 == 0) continue;
        if (arr[i] == true) {
            for (int j=2; i*j<=n; j++) 
                arr[i*j] = false;
            answer++;
        }
    }

    return answer;
}